### DISCUSSION TITLE
Nextcloud AIO v3.2.0 ffmpeg-error: Unrecognized option 'temporal-aq'

### DISCUSSION THREAD

**Discussion by byiigor** [nextcloud/all-in-one#1536] [Questions] [answered]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1.Upgrade to Nextcloud AIO from v3.1 to v3.2.0, or updaete Memories 8x to 9x.
2.Install Memories.
3.Run video through Memories - Videos whith activated transcoding.

### Expected behavior 
Video playing whith transcoding.

### Actual behavior 
Video dont playing. We can see error in /tmp/go-vod.log.

### Host OS 
3.16.3
NAME="Alpine Linux"

#### Nextcloud AIO version 
3.2.0
#### Current channel 
latest
#### Other valuable info 
log:
2022/12/16 15:39:44 sg8csr810ug0-max: /usr/bin/ffmpeg -loglevel warning -hwaccel cuda -hwaccel_output_format cuda -autorotate 0 -i /mnt/ncdata/admin/files/Фото/2021/20210730_201057.MP4 -copyts -vf format=nv12|cuda,hwupload,scale_cuda=passthrough=0 -c:v h264_nvenc -profile:v high -preset p6 -tune ll -temporal-aq 1 -rc vbr -rc-lookahead 30 -cq 24 -c:a aac -ac 1 -b:a 192k -avoid_negative_ts disabled -f hls -hls_time 3 -force_key_frames expr:gte(t,n_forced*3) -hls_segment_type mpegts -start_number 0 -hls_segment_filename /tmp/go-vod/sg8csr810ug0-857476388/max-%06d.ts -
2022/12/16 15:39:44 ffmpeg-error: Unrecognized option 'temporal-aq'.
2022/12/16 15:39:44 ffmpeg-error: Error splitting the argument list: Option not found

Trying to update ffmpeg - ffmpeg version 5.0.1 is used - this is the latest package that available in 3.16.3 Alpine Linux....

Solution for now: remove Memories 9.x and install by hands Memories 8.x.

Any else ideas?

**Comment by szaimen** [comment]
Which ffmpeg verision does memories need?

**Comment by szaimen** [comment]
Based on https://docs.nvidia.com/video-technologies/video-codec-sdk/ffmpeg-with-nvidia-gpu/ it looks like the feature needs a special compiled ffmpeg version? I doubt that this is doable for us...

**Comment by szaimen** [comment]
cc @pulsejet do you know more about this maybe? :)

Please note that it is not possible to add the needed device currently anyway: https://github.com/nextcloud/all-in-one/discussions/1525

**Comment by byiigor** [comment]
I think it is because i dont have nvenc which is only in nvidia cards. And in my nextcloud config.php it was enabled: 'memories.nvenc' => true. 
I'll check it now.

**Reply by byiigor** [reply]
> I think it is because i dont have nvenc which is only in nvidia cards. And in my nextcloud config.php it was enabled: 'memories.nvenc' => true. I'll check it now.

I checked, it was because 'memories.nvenc' => true without nvidia card.  Whith memories.nvenc' => false it works fine.

Possible in  Memories 8.2  'temporal-aq' not used. Thats why ihave no error later.

And you right, "Codec 'h264_nvenc' is not recognized by FFmpeg."  - not built with nvenc support.

**Comment by szaimen** [comment, marked as answer]
I investigated a bit and it looks like ffmpeg is indeed not built with nvenc support. See https://git.alpinelinux.org/aports/tree/community/ffmpeg/APKBUILD?h=3.17-stable

So I fear there is nothing we can do in AIO to fix this as compiling ffmpeg from source is not an option for us.

However you could create an issue upstream to request this feature.

**Reply by Zoey2936** [reply]
an request could be done here: https://gitlab.alpinelinux.org/alpine/aports/-/issues

**Comment by Zoey2936** [comment]
To this: https://github.com/nextcloud/all-in-one/discussions/1525#discussioncomment-4707555 even if there would be an image, I think the problem would be that the dynamic libraries aren't available inside the alpine container... The only good option would be to built it ourselves.

**Reply by szaimen** [reply]
All right, didnt thought about this. Since building ourselves is not an option, this will not be possible.

Thanks for your answer! :)

**Reply by shperrung** [reply]
Hi!
Am I right, thinking that Nextcloud-AIO is "Docker-in-Docker"? 
I dig that awesome project https://github.com/jrottenberg/ffmpeg 
These are images with pre-build ffmpeg for various GPUs. I’m using it form host machine with NVIDIA P400 Quadro calling with recommended command: `docker run --runtime=nvidia jrottenberg/ffmpeg:4.4-nvidia -hwaccel cuvid -c:v h264_cuvid -i INPUT -vf scale_npp=-1:720 -c:v h264_nvenc -preset slow OUTPUT`
May we implement that container into Nextcloud-AIO and use it in same kind instead own ffmpeg in Alpine based image Nextcloud?

**Reply by Zoey2936** [reply]
It is not really docker in docker, AIO just communicates with the docker deamon from the host and for the second, how should this work? Does ffmpeg have something like a demon mode and if yes, does nextcloud support it?

**Reply by shperrung** [reply]
UnfortunatellyI don't know how Nextcloud-AIO is working, but I saw this in compose file `- /var/run/docker.sock:/var/run/docker.sock:ro` I thought that may allow operate another containers from inside Nextcloud.
jrottenberg/ffmpeg:4.4-nvidia - is built from nvidia/cuda:11.4.1-devel-ubuntu20.04 without demon mode.

I tried to bulid my image using JRottenberg's Dockerfile for nvenc based on **nextcloud:25.0.3-apache** image but I faced with a need of **libnpp** library, that is part of **cuda-toolkit** and need a lot of dependencies for ffmpeg.

**Reply by Zoey2936** [reply]
> UnfortunatellyI don't know how Nextcloud-AIO is working, but I saw this in compose file `- /var/run/docker.sock:/var/run/docker.sock:ro` I thought that may allow operate another containers from inside Nextcloud. 

Nextcloud itself has no access to docker, only the mastercontainer can access docker

**Reply by shperrung** [reply]
[Dockerfile.txt](https://github.com/nextcloud/all-in-one/files/10795516/Dockerfile.txt)
Here is my Dockerfile for building apache-nextcloud:25.0.3 with ffmpeg supporting cuda
Inside the container "ffmpeg" and "nvidia-smi" commands return normal response.
I don't know how it will work for alpine-nextcloud but you can test it.

**Reply by Zoey2936** [reply]
> [Dockerfile.txt](https://github.com/nextcloud/all-in-one/files/10795516/Dockerfile.txt) Here is my Dockerfile for building apache-nextcloud:25.0.3 with ffmpeg supporting cuda Inside the container "ffmpeg" and "nvidia-smi" commands return normal response. I don't know how it will work for alpine-nextcloud but you can test it.

Even if it would work, as written above, building ffmpeg from source is not an option... But thanks anyway!
